packages/hash/subgraph/src/types/vertex.ts

```diff
@@ -61,6 +61,6 @@ export type Vertices = {
   };
 } & {
   [_: EntityId]: {
-    [_: EntityVersion]: KnowledgeGraphVertex;
+    [_: Timestamp]: KnowledgeGraphVertex;
```

Comment:
This is a bit of a regression in expressiveness in my mind. 

We could have type-branded `EntityVersion` and then you'd know if you should be able to do a direct lookup for the thing or not. (Timestamps require a search, EntityVersions _should_ correspond to an exact vertex under a certain pinned axis)

I appreciate you've changed `EntityVersion` to now be this complex object thing (which I'm not entirely comfortable with but I don't have a better suggestion right now), maybe there's something else we can call these specific timestamps?